state-sync/storage-service/server/src/metrics.rs

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use diem_metrics::{
+    register_histogram_vec, register_int_counter_vec, HistogramTimer, HistogramVec, IntCounterVec,
+};
+use diem_types::PeerId;
+use network::ProtocolId;
+use once_cell::sync::Lazy;
+use short_hex_str::AsShortHexStr;
+
+/// Counter for pending network events to the storage service (server-side)
+pub static PENDING_STORAGE_SERVER_NETWORK_EVENTS: Lazy<IntCounterVec> = Lazy::new(|| {
+    register_int_counter_vec!(
+        "diem_storage_service_server_pending_network_events",
+        "Counters for pending network events for the storage server",
+        &["state"]
+    )
+    .unwrap()
+});
+
+/// Counter for storage service errors encountered
+pub static STORAGE_ERRORS_ENCOUNTERED: Lazy<IntCounterVec> = Lazy::new(|| {
+    register_int_counter_vec!(
+        "diem_storage_service_server_errors",
+        "Counters related to the storage server errors encountered",
+        &["peer", "protocol", "error_type"]
```

Comment:
Is this the `peer_id` of the server or the `client`?  Mostly, because that would cause a large increase in metrics.  It's probably fine to keep it anonymous to `peer` and then just look at logs to see the specific error / peer.

Reply:
It's fine for testing in say cluster test, but you'll want to prefix the metric name so that they don't collide with the production ones.

I'm pushing to get rid of these peer based metrics in things in like mempool, as in a large network they could expand to a very large size and cause all kinds of problems.